### PR TITLE
Add SUSE staging GPG key to containerized server

### PR DESCRIPTION
**Pull request by nodeg**
## What does this PR change?

This adds the [SUSE staging GPG key](https://build.suse.de/projects/SUSE:ALP:Source:Standard:1.0:Staging:Z/signing_keys) to the containerized server for SUMA

```bash
$ cat SUSE_ALP_Source_Standard_1.0_Staging_Z_key.gpg
-----BEGIN PGP PUBLIC KEY BLOCK-----
Version: GnuPG v2.0.15 (GNU/Linux)

mQENBFErWUoBCADTYW0S3mAHLCLLFuaRWxsK6NwGDmW6vSSkRt8OTXSAJyz2yFUS
oQ7dmw5ce6uyChoMBap0+2xiGyBzMZ4nP3ADD6voG5FXxg6QTfsDyz/sg5BRbS2W
q2e3XU4qEiJn357mXd53YfXKYpA7d2Ct5LKtKGph0DeGsi2zDKnjUCkXrOGNI+VH
qUCardfjAxgsA+m2bN/GKBOdHiHEHlQ0+GyLaoDi8utQz+L42iEQK4uvUQDKa+6k
fbocsR0z55meVCf5ZuEg/katzhtrtesG/sqTLCBMLsLGSIllHSZRxQPbBNf8l7wT
CPqg9JqXHT17aO3O+ZTkqv1dTIwXtmyPkIAjABEBAAG0JVVuc3VwcG9ydGVkIDx1
bnN1cHBvcnRlZC1yc2FAc3VzZS5kZT6JATwEEwECACYCGwMGCwkIBwMCBBUCCAME
FgIDAQIeAQIXgAUCY+utDQUJGkUhwwAKCRCcdTFJzjtnLkZzCACJV8KKLtRtlw5a
MZJdFz1TKFzpAYd2Wcdtl7R/gYa+z4A91UbqtFHdK91EKXAvEkf4f88wj2Dvqj/1
omBhO8si+p3Hxm1UVD91nTBNVWyjn33uai5YP2VEKQOzgBV2swVWuzSsJq0Kc1Yk
zKCPdtFpENdHfRKIXVgaDmUy0EAQJhBLx4SGD9l3DnG5Lt4d+jXe49BgCaGn+5eB
zl23nwL02wSD6uTShsqtdGQCaZtl2WjIb81H6rTRLcSPhyClpy/s3GV9wFri0zLL
Q6eiIbNj8uAvCDdGACEpmXic0lgIQdKsnJLelqJtwCpAmi3ma4cV1kCLD9z2Qtxl
ePb5BMBz
=xYx6
-----END PGP PUBLIC KEY BLOCK-----

$ file SUSE_ALP_Source_Standard_1.0_Staging_Z_key.gpg
SUSE_ALP_Source_Standard_1.0_Staging_Z_key.gpg: PGP public key block Public-Key (old)
```

and fixes

```bash
2024/08/28 10:55:52 +02:00 Command: ['/usr/bin/spacewalk-repo-sync', '--channel', 'custom_channel_slmicro60_minion', '--type', 'yum', '--non-interactive']
2024/08/28 10:55:52 +02:00 Sync of channel started.
2024/08/28 10:55:52 +02:00 RepoMDError: Cannot access repository.
Repository 'custom_channel_slmicro60_minion' is invalid.
[custom_channel_slmicro60_minion|http://download.suse.de/ibs/SUSE:/ALP:/Source:/Standard:/1.0:/Staging:/Z/standard/] Valid metadata not found at specified URL
History:
 - Signature verification failed for repomd.xml

Please check if the URIs defined for this repository are pointing to a valid repository.
Skipping repository 'custom_channel_slmicro60_minion' because of the above error.
Could not refresh the repositories because of errors.
```

where we get our SL Micro 6.0 MIs to test.

Slack discussion: https://suse.slack.com/archives/C02CLBJ5RGF/p1724853951719289

### CODE REVIEW

**Review by Bischoff — APPROVED** [review]

File: salt/server_containerized/testsuite.sls [hunk 1]
```diff
@@ -108,6 +108,25 @@ repo_key_import:
     - name: "mgrctl exec 'rpm --import /tmp/galaxy.key'"
     - onchanges:
       - cmd: galaxy_key_copy
+
+# needed for SL Micro packages coming from staging e.g.
+# https://download.suse.de/ibs/SUSE:/ALP:/Source:/Standard:/1.0:/Staging:/Z/standard/
```
Comment (deneb-alpha) [inline salt/server_containerized/testsuite.sls:113]:
```suggestion
# needed for SL Micro 6.0 maintenance updates coming from staging e.g.
# https://download.suse.de/ibs/SUSE:/ALP:/Source:/Standard:/1.0:/Staging:/Z/standard/
```

**Review by deneb-alpha — COMMENTED** [review]
just a small suggestion on the  comment. feel free to include it  or not :) 

Apart from this, the key is  correct

File: salt/server_containerized/testsuite.sls (hunk 1 above)
Comment (nodeg) [reply salt/server_containerized/testsuite.sls:113]: Thank you, Marina. I will include it.

**Comment by nodeg** [comment]
Hm... I manually added the key and imported it with rpm (basically what the Salt steps in the PR do)

```bash
suma-bv-50-srv:/tmp # mgrctl exec 'rpm --import /tmp/^Cse_staging.key'

suma-bv-50-srv:/tmp # mgrctl cp /tmp/suse_staging.key server:/tmp/suse_staging.key
4:52PM INF Welcome to mgrctl
4:52PM INF Executing command: cp

suma-bv-50-srv:/tmp # mgrctl exec 'rpm --import /tmp/suse_staging.key'
warning: Rebuilding outdated index databases
warning: Generating 18 missing index(es), please wait...
suma-bv-50-srv:/tmp #
```

I then checked the checkbox again, but I still get the same error.

**Comment by mcalmer** [comment]
@nodeg please use the special mgradm command for custome gpg keys

**Comment by maximenoel8** [comment]
Is there any risk to expose this key to public repository ? Should it be a Jenkins secret ?

**Comment by nodeg** [comment]
> @nodeg please use the special mgradm command for custome gpg keys

Thank you, mc. This did the trick:

```bash
suma-bv-50-srv:~ # mgradm gpg add /tmp/suse_staging.key
8:12AM INF Welcome to mgradm
8:12AM INF Executing command: add
8:12AM ??? gpg: keybox '/var/spacewalk/gpg/customer-build-keys.gpg' created
gpg: keybox '/root/.gnupg/pubring.kbx' created
pub   rsa2048 2013-02-25 [SC] [expires: 2027-02-13]
      F5FE65C3163CD347266C583D9C753149CE3B672E
uid                      Unsupported <unsupported-rsa@suse.de>

Do you really want to trust this key [y/N]?y
8:12AM INF Running gpg --no-default-keyring --import --import-options import-minimal --keyring /var/spacewalk/gpg/customer-build-keys.gpg /var/spacewalk/gpg/suse_staging.key
8:12AM ??? gpg: key 9C753149CE3B672E: public key "Unsupported <unsupported-rsa@suse.de>" imported
8:12AM ??? gpg: Total number processed: 1
gpg:               imported: 1
8:12AM INF Running systemctl restart uyuni-update-config
```

```bash
2024/08/29 08:13:05 +02:00 Command: ['/usr/bin/spacewalk-repo-sync', '--channel', 'custom_channel_slmicro60_minion', '--type', 'yum', '--non-interactive']
2024/08/29 08:13:05 +02:00 Sync of channel started.
2024/08/29 08:13:06 +02:00 Repo URL: http://download.suse.de/ibs/SUSE:/ALP:/Source:/Standard:/1.0:/Staging:/Z/images/repo/SUSE-Manager-Tools-For-SL-Micro-6-x86_64/
2024/08/29 08:13:06 +02:00     Packages in repo:                 9
2024/08/29 08:13:06 +02:00     No new packages to sync.
2024/08/29 08:13:06 +02:00 Filtering packages that failed to download
2024/08/29 08:13:06 +02:00 Importing packages started.
2024/08/29 08:13:06 +02:00 
2024/08/29 08:13:06 +02:00   Importing packages to DB:
2024/08/29 08:13:06 +02:00 Importing packages finished.
2024/08/29 08:13:06 +02:00 
2024/08/29 08:13:06 +02:00   Patches in repo: 1.
2024/08/29 08:13:06 +02:00     No new patch to sync.
2024/08/29 08:13:06 +02:00 
2024/08/29 08:13:06 +02:00   Importing mediaproducts file products.
2024/08/29 08:13:06 +02:00 *** NOTE: Importing mediaproducts file for the channel 'custom_channel_slmicro60_minion'. Previous mediaproducts will be discarded.
2024/08/29 08:13:06 +02:00   Updating overview of 1 systems
2024/08/29 08:13:06 +02:00 Sync completed.
```

I am not sure, why we use the command above for the other keys, we import, e.g.
https://github.com/uyuni-project/sumaform/blob/22f1f1d8e435d339727693416a1c56a1a2aa2325/salt/server_containerized/testsuite.sls#L76-L111

**Comment by mcalmer** [comment]
> I am not sure, why we use the command above for the other keys, we import, e.g.

I think this was added before this command exists?

**Comment by nodeg** [comment]
That makes sense. Then I will adjust the other comands, too.

**Comment by nodeg** [comment]
> Is there any risk to expose this key to public repository ? Should it be a Jenkins secret ?

I do not know. However, what I added above is what we have been doing before. Since this could also be interesting for non Jenkins deployments e.g. on test servers, personal environments, I would not add it as a Jenkins secret.

File: salt/server_containerized/testsuite.sls [hunk 2]
```diff
@@ -79,35 +79,36 @@ uyuni_key_copy_host:
...
-uyuni_key_copy:
-  cmd.run:
-    - name: "mgrctl cp /tmp/uyuni.key server:/tmp/uyuni.key"
-    - onchanges:
-      - file: uyuni_key_copy_host
-
 repo_key_import:
   cmd.run:
-    - name: "mgrctl exec 'rpm --import /tmp/uyuni.key'"
+    - name: "mgradm gpg add /tmp/uyuni.key"
```
Comment (szachovy) [inline salt/server_containerized/testsuite.sls:84]: Do we need that once or more times? `/tmp` is considered volatile and its contents are may be cleared out on reboot or at regular intervals by the operating system to free up space.

**Review by ktsamis — APPROVED** [review]

File: salt/server_containerized/testsuite.sls (hunk 2 above)
Comment (Bischoff) [reply salt/server_containerized/testsuite.sls:84]: Wiktor, I think that `/tmp` is used here only as a transit area. The state right above takes care of the copy into that transit area.

Comment (nodeg) [reply salt/server_containerized/testsuite.sls:84]: So before, we copied it over to the container to be able to import it with `rpm`. However, with the new `mgradm gpg` command, the tool does this, including importing the new key to the container OS automatically.
Please correct me if I am wrong @mbussolotto @mcalmer.

Comment (nodeg) [reply salt/server_containerized/testsuite.sls:84]: @Bischoff: Yes. The code above

```sls
uyuni_key_copy_host:
  file.managed:
    - name: /tmp/uyuni.key
    - source: salt://default/gpg_keys/uyuni.key
```

copies the key "from Salt" to `'tmp`

Comment (mcalmer) [reply salt/server_containerized/testsuite.sls:84]: yes

**Review by mcalmer — APPROVED** [review]

File: salt/server_containerized/testsuite.sls (hunk 2 above)
Comment (nodeg) [reply salt/server_containerized/testsuite.sls:84]: Thank you for the confirmation, mc.

**Comment by mcalmer** [comment]
> Is there any risk to expose this key to public repository ? Should it be a Jenkins secret ?

It is a "public key". So I think it is not a problem as the private part is only internal. 
And if you require it for testing, we need to have it

File: salt/server_containerized/testsuite.sls (hunk 2 above)
Comment (szachovy) [reply salt/server_containerized/testsuite.sls:84]: ok, closing

**Review by szachovy — APPROVED** [review]

**Pull request merged**